File: prob2.c
/* Calculate all of the even fibonaci numbers under 4 million */
#include <stdio.h>
main(){
  int fibNumMinusOne = 0;
  int fibNum = 1;
  int sum = 0;
  int temp= 0;
  while(fibNum<4000000){ // its a magic number but its a small problem
    if(fibNum%2==0){
      sum+=fibNum;
    }
    //here's the loop method to get fib of n+1
    temp = fibNumMinusOne;
    fibNumMinusOne = fibNum;
    fibNum+=temp;
  }
  printf("%d \n",sum);
}

File: prob2.py
# This calculates the sum of all the even fibonacci numbers less than 4 million
def Sumfib(n):
  """This method takes int n and returns the sum of the fib nums from fib(1) to fib(n) """
    sum = 0 # accumulator pattern
    fibNMinusOne = 0
    fibN = 1
    while fibN < n: #a loop uses way less memory than recusive calls
        if fibN%2==0:
            sum+=fibN
        temp = fibN
        fibN+=fibNMinusOne
        fibNMinusOne = temp
    return sum

print(Sumfib(4000000))
